app/models/setting.rb

```diff
@@ -0,0 +1,19 @@
+# == Schema Information
+#
+# Table name: settings
+#
+#  id          :bigint           not null, primary key
+#  description :string           default("")
+#  key         :string           not null
+#  value       :string
+#
+
+class Setting < ApplicationRecord
+  SUCCESS_PREFIX = 'success_rate'.freeze
+
+  validates :key, uniqueness: true, presence: true
+
+  scope :success_rate, lambda { |department_name, metric_name|
+    where(key: "#{SUCCESS_PREFIX}_#{department_name}_#{metric_name}")
```

Comment:
I think that instead of this "dynamic" key, you could have had polymorphic associations. It'd be a more relational solution, anyways I guess this is fine for a setting 🤔